Apps/Contoso.WPF.Demo/App.xaml.cs

```diff
@@ -13,10 +14,84 @@ namespace Contoso.WPF.Demo
     /// </summary>
     public partial class App
     {
+        private const string LogTag = "AppCenterDemo";
+
         protected override void OnStartup(StartupEventArgs e)
         {
             AppCenter.LogLevel = LogLevel.Verbose;
+            Crashes.ShouldAwaitUserConfirmation = ConfirmationHandler;
+            Crashes.ShouldProcessErrorReport = ShouldProcess;
+
+            // Event handlers.
+            Crashes.SendingErrorReport += SendingErrorReportHandler;
+            Crashes.SentErrorReport += SentErrorReportHandler;
+            Crashes.FailedToSendErrorReport += FailedToSendErrorReportHandler;
             AppCenter.Start("f4e2a83d-3052-4884-8176-8b2c50277d16", typeof(Analytics), typeof(Crashes));
+            Crashes.HasCrashedInLastSessionAsync().ContinueWith(hasCrashed =>
+            {
+                AppCenterLog.Info(LogTag, "Crashes.HasCrashedInLastSession=" + hasCrashed.Result);
+            });
+            Crashes.GetLastSessionCrashReportAsync().ContinueWith(task =>
+            {
+                AppCenterLog.Info(LogTag, "Crashes.LastSessionCrashReport.Exception=" + task.Result?.Exception);
+            });
+        }
+
+        bool ShouldProcess(ErrorReport report)
+        {
+            AppCenterLog.Info(LogTag, "Determining whether to process error report");
+            return true;
```

Comment:
Better to have a UI to set this value for testing. Add a UI component and have either `true` or `false` that could be selected by testers, persist it in the application and read the value in the method.
I'm not sure this isn't in scope but this will allow us to test ShouldProcess that returns both `true` and `false`.

Reply:
This is aligning with other platforms but probably a good to have one for testing in the future.